app/routes.py

```diff
@@ -0,0 +1,269 @@
+    video = Video.query.get(video_id)
+    # check if customer and video exist
+    if customer is None or video is None:
+        return make_response("", 404)
+    num_of_videos_rented = Rental.query.filter_by(
+        video_id=video.id, checked_out=True).count()
+    available_inventory = video.total_inventory - num_of_videos_rented
+    if available_inventory == 0:
+        return jsonify({"message": f"Could not perform checkout"}), 400
+    new_rental = Rental(customer_id=customer.id,
+                        video_id=video.id,
+                        due_date=(datetime.now() + timedelta(days=7)),
+                        checked_out=True)
+    db.session.add(new_rental)
+    db.session.commit()
+    videos_checkout_by_customer = Rental.query.filter_by(
+        customer_id=customer.id, checked_out=True).count()
+    available_inventory -= 1
+    return jsonify({
+        "customer_id": new_rental.customer_id,
+        "video_id": new_rental.video_id,
+        "due_date": new_rental.due_date,
+        "videos_checked_out_count": videos_checkout_by_customer, "available_inventory": available_inventory}), 200
+
+
+# CHECK IN A VIDEO TO A CUSTOMER
+@rentals_bp.route("/check-in", methods=["POST"])
+def checkin_video():
+    request_body = request.get_json()
+    if "customer_id" not in request_body:
+        return {"details": "Request body must include customer_id."}, 400
+    elif "video_id" not in request_body:
+        return {"details": "Request body must include video_id."}, 400
+    customer_id = request_body["customer_id"]
+    video_id = request_body["video_id"]
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+    # check if customer and video exist
+    if customer is None or video is None:
+        return make_response("", 404)
+    rental = Rental.query.filter_by(
+        video_id=video.id, customer_id=customer.id, checked_out=True).first()
+    if rental is None:
+        return jsonify({"message": f"No outstanding rentals for customer {customer.id} and video {video.id}"}), 400
+    rental.checked_out = False
+    db.session.commit()
+    num_of_videos_rented = Rental.query.filter_by(
+        video_id=video.id, checked_out=True).count()
+    available_inventory = video.total_inventory - num_of_videos_rented
+    videos_checkout_by_customer = Rental.query.filter_by(
+        customer_id=customer.id, checked_out=True).count()
+    return jsonify({
+        "customer_id": rental.customer_id,
+        "video_id": rental.video_id,
+        "videos_checked_out_count": videos_checkout_by_customer, "available_inventory": available_inventory}), 200
+
+
+# GET RENTALS BY CUSTOMER
+@customers_bp.route("/<id>/rentals", methods=["GET"])
+def rentals_by_video(id):
+    customer = Customer.query.get(id)
+    if customer is None:
+        return jsonify({"message": f"Customer {id} was not found"}), 404
+    rentals = Rental.query.filter_by(
+        customer_id=customer.id, checked_out=True).all()
```

Comment:
You've done a good job of remembering to add the `checked_out=True` in your queries, but notice how easy it would be to forget, or if there was a new dev on your team, it might not occur to them they need it. Moving even a single complex line into a well-named helper function can be very useful for avoiding errors.